src/main/java/org/cbioportal/web/util/StudyViewFilterUtil.java

```diff
@@ -53,12 +40,12 @@ public StudyViewFilterUtil() {
         geneService = null;
     }
 
-    public void extractStudyAndSampleIds(
-        List<SampleIdentifier> sampleIdentifiers,
+    public <T extends SampleIdentifier> void extractStudyAndSampleIds(
```

Comment:
it's extracting from the `sampleIdentifiers` parameter. i think this method was done this way because they wanted to populate two lists with a single pass of `sampleIdentifiers` instead of two. i'm reluctant to change it if thats how it worked before and i would have to update several usages of this. let me know if you want me to still update this